.gitignore

```diff
@@ -25,6 +25,8 @@ CMakeFiles
 *.o
 *.obj
 compile_commands.json
+**/Output
```

Comment:
Just `/Output` should be enough, no?

Reply:
No, `lit` actually sprinkles `Output` directories all over the place. After running the tests in this PR locally with an in-tree build I have

```
./test/Output
./test/validation/Output
./test/wasm-emscripten-finalize/Output
```

Reply:
Cluttering test directory does not sound very ideal.. Can we put all of them under out/ or somewhere? There seems to be an option called `test_exec_root` in `llvm-lit` with which we can control the output directory. Can we use it?

Reply:
These files get written to the build directory, so it only clutters your source tree if your do "in-tree" building.. which I think most of us dont.

If you do do "in-tree" building your source tree is already full of clutter.

Reply:
If we banned in-tree building we could remove all these things from gitignore.. which I would personally be an favor of.     I personally like it when git status tells me about stray `.o` files in my checkout, but these rules prevent that.